lib/fileset/tests.sh

```diff
@@ -118,7 +118,7 @@ expectFailure() {
         die "$expr evaluated successfully to $result, but it was expected to fail"
     fi
     stderr=$(<"$tmp/stderr")
-    if [[ ! "$stderr" =~ $expectedErrorRegex ]]; then
+    if ! grep -E "$expectedErrorRegex" <<< $stderr; then
```

Comment:
Unfortunately this only seems to check the first line of the regex. It doesn't seem like `grep` supports that. So I think we should stick with the bash regex match for now.